src/utils/streams.js

```diff
@@ -0,0 +1,115 @@
+var fs = require('fs');
+var csv = require('csvtojson');
+var program = require('commander');
+
+var reader = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+    terminal: false
+});
+
+function validateOnDefinitions(filePath) {
+    if (!filePath) {
+        console.log(`File path - ${filePath} is undefined!`);
+        process.exit(1);
+    }
+}
+
+function validateFileExtension(filePath) {
+    var regex = new RegExp('.+(\.csv)$', 'i');
+    if (!regex.test(filePath)) {
+        console.log(`File path - ${filePath} doesn't valid!`);
+        process.exit(1);
+    }
+}
+
+function validateOnFileExist(filePath) {
+    if (!fs.existsSync(filePath)) {
+        console.log(`File - ${filePath} doesn't exist!`);
+        process.exit(1);
+    }
+}
+
+function handleIOError(error) {
+    console.error("Iccurred an error during working with files", error.message);
+    process.exit(1);
+}
+
+function reverse(str) {
+    reader.question("Please enter the text: ", function (data) {
+        console.log(`Reversed data: ${data.split('').reverse().join('')}`);
+        reader.close();
+    });
+}
+
+function transform(str) {
+    reader.question("Please enter the text: ", function (data) {
+        console.log(`Uppered case data: ${data.toUpperCase()}`);
+        reader.close();
+    });
+}
+
+function outputFile(filePath) {
+    validateOnDefinitions(filePath);
+    validateOnFileExist(filePath);
+    fs.createReadStream(filePath)
+        .on('error', handleIOError)
+        .pipe(process.stdout);
+}
+
+function convertFromFile(filePath) {
+    validateOnDefinitions(filePath);
+    validateOnFileExist(filePath);
+    validateFileExtension(filePath);    
+    fs.createReadStream(filePath)
+        .on('error', handleIOError)
+        .pipe(csv())
+        .pipe(process.stdout);
+}
+
+function convertToFile(filePath) {
+    validateOnDefinitions(filePath);
+    validateOnFileExist(filePath);
+    validateFileExtension(filePath);    
+    fs.createReadStream(filePath)
+        .on('error', handleIOError)
+        .pipe(csv())
+        .pipe(fs.createWriteStream(filePath.replace(/(.*)\.csv$/, '$1.json')));
+}
+
+function checkArgs(args) {
+    const firstArg = args[2];
+    if (!firstArg) {
+        args[2] = '--help';
+    } else if (['-a', '--action'].includes(firstArg)) {
+        args.forEach((element, index, array) => {
+            if (['-h', '--help'].includes(element)) {
+                array[index] = '';
+            }
+        });
+    }
+}
+
+program
+    .version('1.0.0', '-v, --version')
+    .option('-a, --action <action>', 'Function to run: <reverse|transform|outputFile|convertFromFile|convertToFile>',
+        /^(reverse|transform|outputFile|convertFromFile|convertToFile)$/i)
+    .option('-f, --file <filePath>', 'Full file name to pass to the functions outputFile|convertFromFile|convertToFile');
+
+checkArgs(process.argv);
+
+program.parse(process.argv);
+
+switch (program.action) {
```

Comment:
try to avoid usage of swicth/case conditions.
hint: in JS you can create an object with shorthand syntax like
```
const commands = {
  reverse, // instead of reverse: reverse,
  transform
  etc.,......
}
```
And instead of switch/case then access through
`commands[program.action](program.file || program.path)`